Sources/FoundationEssentials/UUID.swift

```diff
@@ -23,26 +17,29 @@ public typealias uuid_string_t = (Int8, Int8, Int8, Int8, Int8, Int8, Int8, Int8
 public struct UUID : Hashable, Equatable, CustomStringConvertible, Sendable {
     public private(set) var uuid: uuid_t = (0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)
 
+    public enum Version {
+        case v4
+    }
+
     /* Create a new UUID with RFC 4122 version 4 random bytes */
-    public init() {
-        withUnsafeMutablePointer(to: &uuid) {
-            $0.withMemoryRebound(to: UInt8.self, capacity: MemoryLayout<uuid_t>.size) {
-                uuid_generate_random($0)
-            }
+    public init(version: Version = .v4) {
+        switch version {
+            case .v4:
+                self.uuid = Self.v4_generatedRandom()
         }
     }
 
     /// Create a UUID from a string such as "E621E1F8-C36C-495A-93FC-0C247A3E6E5F".
     ///
     /// Returns nil for invalid strings.
-    public init?(uuidString string: __shared String) {
-        let res = withUnsafeMutablePointer(to: &uuid) {
-            $0.withMemoryRebound(to: UInt8.self, capacity: 16) {
-                return uuid_parse(string, $0)
-            }
-        }
-        if res != 0 {
-            return nil
+    public init?(version: Version = .v4, uuidString string: __shared String) {
```

Comment:
This would be an ABI breakage, since we're removing the previous initializer.

Reply:
That makes sense, thanks for letting me know Tony. 

I restored the original initializers in c3cc89cc5faf0bbd36c9123a73e996cab405569c, however I also understand that the changes in this PR are something that the team aren't ready to look at yet.